LivlReviewsApi/Repositories/EntityRepository.cs

```diff
@@ -0,0 +1,100 @@
+using LivlReviewsApi.Data;
+using Microsoft.EntityFrameworkCore;
+
+namespace LivlReviewsApi.Repositories;
+
+public class EntityRepository<T> : IRepository<T> where T : class
+{
+    private readonly AppDbContext DbContext;
+    private readonly DbSet<T> DbSet;
+    
+    public EntityRepository(AppDbContext dbContext)
+    {
+        DbContext = dbContext;
+        DbSet = DbContext.Set<T>();
+    }
+    public T GetById(int id)
+    {
+        return DbSet.Find(id);
+    }
+
+    public List<T> GetAll()
+    { 
+        return DbSet.ToList();
+    }
+    
+    public List<T> GetBy(Func<T, bool> predicate)
+    {
+        return DbSet.Where(predicate).ToList();
+    }
+
+    public PaginatedResult<T> GetPaginated(Func<T, bool> predicate , PaginationParameters paginationParameters)
```

Comment:
We can create an PaginatedEntityRepository to manage paginated entities, to not have theses functions when we want to manage non paginated entities